### ISSUE TITLE
Create new user by SSO without allowing email self-registrations.

### ISSUE THREAD

**Issue by jonthorne** [humhub/humhub#6780]
### Is your feature request related to a problem? Please describe.

Using SSO login with a valid user in Keycloak, but not yet in humhub and "New users can register" is disabled, this blocks the new user registration. I only want Keycloak users to be able to login and don't want others to be able to self register by email.

### Describe the solution you'd like

I would like a way to disable user registration on humhub via the "register form" while still allowing the SSO flow to provision users if they do not exists in humhub. I want to control the registration of new users via another system, not on humhub. And allow any valid Keycloak user to be able to login to hubhub.

### Describe alternatives you've considered

The only option I can find for now is to enable "New users can register". But try and find a way to block registration by email account.

**Comment by ArchBlood** [comment]
In your theme view overwrite you can remove the following lines as alternative;

https://github.com/humhub/humhub/blob/56d42191afecf4c06c45614f9d154dc67f6c1889/protected/humhub/modules/user/views/auth/login.php#L99-L113

https://github.com/humhub/humhub/blob/56d42191afecf4c06c45614f9d154dc67f6c1889/protected/humhub/modules/user/views/auth/login_modal.php#L110-L125

**Comment by jonthorne** [comment]
Hi @ArchBlood

Thanks for this prompt reply. This is a very helpful way to get around the issue for now. Although I will have to make sure to keep modifying this after any software update. A separate setting for this scenario would still be helpful if at all possible.

Jon.

**Comment by marc-farre** [comment]
I use CSS to deal with this problem:
1. You need Humhub 1.16 or to patch your 1.15 with this: https://github.com/humhub/humhub/pull/6616/files
2. Add this CSS code to your custom theme:

```css
#user-auth-login-modal #tabs,
#create-account-form[data-has-auth-client="0"] #registration-form,
form[action="/user/auth/login"],
#register-form,
#account-login-form,
.or-container {
  display: none !important;
}
```

See comment on https://community.humhub.com/comment/perma?id=46424 (You must be a member of the https://community.humhub.com/s/installation-and-setup space).

You can also skip the login form this way (won't work for account creation):
https://community.humhub.com/s/installation-and-setup/wiki/Configuration+Examples#skip-login-form

Maybe we could leave this issue open to provide later a better solution.

**Comment by luke-** [comment]
We could either add a new checkbox "New user can register via SSO" or create exceptions for SSO providers via the configuration file.

**Comment by marc-farre** [comment]
> We could either add a new checkbox "New user can register via SSO" or create exceptions for SSO providers via the configuration file.

I think the idea would be to have a "Disable basic authentification" checkbox, or something like that, to disable the possibility to create an account or login with the Humhub native login form. This will allow to force using SSO.

**Comment by luke-** [comment]
As a fallback, we should always allow a login via the HumHub login form.

Some SSO modules such as SAML or SSO already offer the option to skip the login form. But they also provide a fallback option.


If I have understood correctly, the main issue is that registration via e-mail should not be possible, but new registrations should be made automatically via SSO.

**Comment by marc-farre** [comment]
Yes. Maybe we could hide the basic login form via a setting in the `protected/config/common.php` file?
This way, we don't need any fallback (which is confusing for users how don't remember which way they should login), because in case of problem we only have to comment this setting in the file to display again the basic login form.

**Comment by luke-** [comment]
Would this https://github.com/humhub/humhub/pull/6788 PR solves the issue?
